django/contrib/auth/models.py

```diff
@@ -357,6 +357,17 @@ def get_short_name(self):
         "Returns the short name for the user."
         return self.first_name
 
+    def get_hash_value(self, timestamp):
+        """Returns the timestamped hash value for purposes of generating
+        a password reset token.
+        """
+        # Ensure results are consistent across DB backends
+        login_timestamp = '' if self.last_login is None else self.last_login.replace(microsecond=0, tzinfo=None)
+        return (
+            six.text_type(self.pk) + self.password + self.email +
```

Comment:
Since `AbstractUser` is subclass of `AbstractBaseUser`, perhaps we could reuse `AbstractBaseUser.get_hash_value()`:

```py
def get_hash_value(self, timestamp):
    hash_value = super(AbstractUser, self).get_hash_value(timestamp)
    return hash_value + self.email
```